cloudbuild.yaml

```diff
@@ -27,16 +27,25 @@ steps:
       terraform init -no-color
       terraform validate -no-color
     dir: 'applications/ray/'
-    waitFor: ['-']
+    waitFor: ['validate platform']
   
   - id: 'validate jupyterhub'
     name: 'gcr.io/$PROJECT_ID/terraform'
     script: |
       terraform init -no-color
       terraform validate -no-color
     dir: 'applications/jupyter/'
-    waitFor: ['-']
+    waitFor: ['validate platform']
+  
+  - id: 'validate rag'
+    name: 'gcr.io/$PROJECT_ID/terraform'
+    script: |
+      terraform init -no-color
+      terraform validate -no-color
+    dir: 'applications/rag/'
+    waitFor: ['validate platform']
 
+  # Create cluster to test ray, jupyterhub
   - id: 'create gke cluster'
```

Comment:
Doing the tests & cleanup in sequence per application instead of complete cleanup at the end. So before rag is tested, ray & jupyter deployments are cleaned up.